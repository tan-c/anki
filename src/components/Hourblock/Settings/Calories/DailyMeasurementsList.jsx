import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import moment from 'moment-timezone';

import { DailyMeasurementActions } from 'utility-redux/dailyMeasurement';
import InputNewConnected from 'utility-react-component/Form/Input/New';
import InputUncontrolledConnected from 'utility-react-component/Form/Input/Uncontrolled';
import DailyCaloriesChartConnected from './DailyCaloriesChart';

export class DailyMeasurementsList extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const { dailyMeasurements, todayDate } = this.props;
    const fieldList = ['weight', 'height', 'waist', 'armCircumference', 'thighCircumference'];

    return (
      <React.Fragment>
        <section className="flex-1">
          <div className="section-header">
            Daily Measurement Item List
          </div>
          <div className="section-content">
            <div className="list-with-pinned-bottom">
              <span className="flex-container-row">
                <span className="flex-2" />
                {fieldList.map(field => <span key={field} className="flex-1">{field}</span>)}
              </span>

              {dailyMeasurements.valueSeq().map(dailyMeasurement => (
                <div className="flex-container-row typical-setup" key={dailyMeasurement.get('_id')}>
                  <span className="flex-2">
                    {moment(dailyMeasurement.get('startedAt')).format('MM-DD')}
                  </span>
                  {fieldList.map(field => (
                    <InputUncontrolledConnected
                      key={field}
                      inputName={field}
                      inputClassNames="flex-1"
                      record={dailyMeasurement}
                      actions={this.props.DailyMeasurementActions}
                    />
                  ))}
                </div>))}
            </div>
            <div className="flex-container-row pinned-bottom border-top">
              <InputNewConnected
                newRecord={{
                  startedAt: todayDate,
                }}
                inputClassNames="flex-1"
                actions={this.props.DailyMeasurementActions}
              />
            </div>
          </div>
        </section>
        <section className="flex-1">
          <DailyCaloriesChartConnected />
        </section>
      </React.Fragment>
    );
  }
}

DailyMeasurementsList.defaultProps = {
};

DailyMeasurementsList.propTypes = {
  todayDate: PropTypes.object.isRequired,
  dailyMeasurements: PropTypes.object.isRequired,
  DailyMeasurementActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    todayDate: moment().tz('Asia/Tokyo').startOf('day').toDate(),
    dailyMeasurements: state.dailyMeasurements,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    DailyMeasurementActions: bindActionCreators(DailyMeasurementActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(DailyMeasurementsList);
