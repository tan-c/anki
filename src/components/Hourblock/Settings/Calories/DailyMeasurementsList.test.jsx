import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

import { Map } from 'immutable';
import moment from 'moment-timezone';

import { DailyMeasurementsList } from './DailyMeasurementsList';

function setup() {
  const props = {
    dailyMeasurements: Map(),
    todayDate: moment.tz('2017-01-01', 'Asia/Tokyo'),

    DailyMeasurementActions: {},
  };

  return shallow(<DailyMeasurementsList {...props} />);
}

describe('DailyMeasurementsList Unit Tests', () => {
  describe('', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      expect(jsonOutput).toMatchSnapshot();
      // expect(wrapper.find("div[data-role='dailymeasurements-list']")).toHaveLength(1);
    });
  });
});
