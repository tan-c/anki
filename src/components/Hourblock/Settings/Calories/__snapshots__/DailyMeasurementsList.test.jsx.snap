// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyMeasurementsList Unit Tests  renders correctly 1`] = `
<Fragment>
  <section
    className="flex-1"
  >
    <div
      className="section-header"
    >
      Daily Measurement Item List
    </div>
    <div
      className="section-content"
    >
      <div
        className="list-with-pinned-bottom"
      >
        <span
          className="flex-container-row"
        >
          <span
            className="flex-2"
          />
          <span
            className="flex-1"
            key="weight"
          >
            weight
          </span>
          <span
            className="flex-1"
            key="height"
          >
            height
          </span>
          <span
            className="flex-1"
            key="waist"
          >
            waist
          </span>
          <span
            className="flex-1"
            key="armCircumference"
          >
            armCircumference
          </span>
          <span
            className="flex-1"
            key="thighCircumference"
          >
            thighCircumference
          </span>
        </span>
      </div>
      <div
        className="flex-container-row pinned-bottom border-top"
      >
        <Connect(InputNew)
          actions={Object {}}
          inputClassNames="flex-1"
          newRecord={
            Object {
              "startedAt": "2016-12-31T15:00:00.000Z",
            }
          }
        />
      </div>
    </div>
  </section>
  <section
    className="flex-1"
  >
    <Connect(DailyCaloriesChart) />
  </section>
</Fragment>
`;
